xayn-ai/src/reranker/public.rs

```diff
@@ -141,108 +140,59 @@ impl<SV, SM, QAV, QAM, DM> Builder<SV, SM, QAV, QAM, DM> {
-            domain: self.domain,
-        }
-    }
-
-    pub fn with_smbert_from_file(
-        self,
-        vocab: impl AsRef<Path>,
-        model: impl AsRef<Path>,
-    ) -> Result<Builder<impl BufRead, impl Read, QAV, QAM, DM>, Error> {
-        Ok(Builder {
-            database: self.database,
-            smbert: SMBertBuilder::from_files(vocab, model)?,
-            qambert: self.qambert,
-            domain: self.domain,
-        })
-    }
-
-    pub fn with_qambert_from_reader<W, N>(self, vocab: W, model: N) -> Builder<SV, SM, W, N, DM> {
-        Builder {
-            database: self.database,
-            smbert: self.smbert,
-            qambert: QAMBertBuilder::new(vocab, model),
-            domain: self.domain,
-        }
-    }
-
-    pub fn with_qambert_from_file(
-        self,
-        vocab: impl AsRef<Path>,
-        model: impl AsRef<Path>,
-    ) -> Result<Builder<SV, SM, impl BufRead, impl Read, DM>, Error> {
-        Ok(Builder {
-            database: self.database,
-            smbert: self.smbert,
-            qambert: QAMBertBuilder::from_files(vocab, model)?,
-            domain: self.domain,
-        })
-    }
-
-    pub fn with_domain_from_reader<N>(self, model: N) -> Builder<SV, SM, QAV, QAM, N> {
+    pub fn with_domain_from_reader<N>(self, model: N) -> Builder<'a, SP, QP, N> {
         Builder {
             database: self.database,
-            smbert: self.smbert,
-            qambert: self.qambert,
+            smbert_config: self.smbert_config,
+            qambert_config: self.qambert_config,
             domain: DomainRerankerBuilder::new(model),
         }
     }
 
     pub fn with_domain_from_file(
         self,
         model: impl AsRef<Path>,
-    ) -> Result<Builder<SV, SM, QAV, QAM, impl Read>, Error> {
+    ) -> Result<Builder<'a, SP, QP, impl Read>, Error> {
         Ok(Builder {
             database: self.database,
-            smbert: self.smbert,
-            qambert: self.qambert,
+            smbert_config: self.smbert_config,
+            qambert_config: self.qambert_config,
             domain: DomainRerankerBuilder::from_file(model)?,
         })
     }
 
     pub fn build(self) -> Result<Reranker, Error>
     where
-        SV: BufRead,
-        SM: Read,
-        QAV: BufRead,
-        QAM: Read,
         DM: Read,
     {
-        let database = self.database;
-        let smbert = SMBert::from(Arc::new(
-            self.smbert
-                .with_token_size(52)?
-                .with_accents(false)
-                .with_lowercase(true)
-                .with_pooling(AveragePooler)
-                .build()?,
-        ));
-        let qambert = self
-            .qambert
+        let smbert_config = self
+            .smbert_config
+            .with_token_size(52)?
+            .with_accents(false)
+            .with_lowercase(true)
+            .with_pooling(AveragePooler);
+
+        let smbert_pipeline = rubert::Pipeline::from(smbert_config)?;
```

Comment:
what about
```suggestion
        let smbert_pipeline = rubert::SMBert::from(smbert_config)?;
```

or we use `As`
```rust
use rubert::{AveragePooler, QAMBert, QAMBertConfig, SMBert, SMBertConfig};

use crate::embedding::smbert::SMBert As ArcSMBert;

let smbert_pipeline = SMBert::from(smbert_config)?;
```

Reply:
See comment above. I don't see the benefit, from the code reader's perspective, of introducing an alias here. Can you elaborate why this was decided in the past?